File: 백준/Gold/1117. 색칠 1/색칠 1.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        long W = Long.parseLong(st.nextToken());
        long H = Long.parseLong(st.nextToken());
        long f = Long.parseLong(st.nextToken());
        long c = Long.parseLong(st.nextToken());
        long x1 = Long.parseLong(st.nextToken());
        long y1 = Long.parseLong(st.nextToken());
        long x2 = Long.parseLong(st.nextToken());
        long y2 = Long.parseLong(st.nextToken());

        long fEnd = f > W / 2 ? W - f : f;
        long total = W * H;
        long colored = (x2 - x1) * (y2 - y1);
        long fWrapped;

        if (x1 >= fEnd) {
            fWrapped = 0;
        } else if (fEnd >= x2)  {
            fWrapped = colored;
        } else {
            fWrapped = (fEnd - x1) * (y2 - y1);
        }

        c++;

        long result = total - (fWrapped * 2 + (colored - fWrapped)) * c;
        System.out.println(result);
    }
}

File: 백준/Gold/1117. 색칠 1/색칠 1.py
# 색칠1
import sys
input = sys.stdin.readline

W, H, f, c, x1, y1, x2, y2 = map(int,input().split())

if f > W // 2:
    f_end = W - f
else:
    f_end = f

total = W * H
colored = (x2 - x1) * (y2 - y1)

if x1 >= f_end:
    f_wrapped = 0
elif x1 < f_end < x2:
    f_wrapped = (f_end - x1) * (y2 - y1)
else:
    f_wrapped = colored

c += 1

result = total - (f_wrapped * c * 2 + (colored - f_wrapped) * c)

print(result)
